### PR TITLE
Add python grpcExtension module

### PR DESCRIPTION
**Pull request by nicolasbotto**

### CODE REVIEW

**Comment by naiteeks** [comment]
@marc19997  - can you review this PR

File: src/edge/modules/grpcExtension/app/batchImageProcessor.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        totalColor = cvGrayImage.sum()
+        avgColor = totalColor / len(grayBytes)
+        colorIntensity = 'dark'
+
+        if(avgColor < 127):
+            colorIntensity = 'light'
+
+        logging.info('Color intensity: {}'.format(colorIntensity))
+        
+        inference = mediaStreamMessage.media_sample.inferences.add()
```
Comment (marc19997) [inline src/edge/modules/grpcExtension/app/batchImageProcessor.py:36]: why can't we just create a direct Inference class from the protobuf file?

File: src/edge/modules/grpcExtension/app/inference_server.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        try:
+            # media descriptor holding input data format
+            self._mediaStreamDescriptor = mediaStreamDescriptor
+
+            # Get how data will be transferred
+            if self._mediaStreamDescriptor.WhichOneof("data_transfer_properties") is None:
+                self._contentTransferType = TransferType.BYTES
+            elif self._mediaStreamDescriptor.HasField("shared_memory_buffer_transfer_properties"):
+                self._contentTransferType = TransferType.REFERENCE
+            elif self._mediaStreamDescriptor.HasField("shared_memory_segments_transfer_properties"):
```
Comment (marc19997) [inline src/edge/modules/grpcExtension/app/inference_server.py:42]: This is not a public data_transfer_property its internal to msft

File: src/edge/modules/grpcExtension/app/inference_server.py [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+                    messageCount += 1
+                else:
+                    # Process batch
+                    logging.info('Processing batch ({0}).'.format(messageCount))
+                    mediaStreamMessage = extension_pb2.MediaStreamMessage()
+                    for image in imageBatch:
+                        mediaStreamMessage = self.ProcessMediaSample(mediaStreamMessage, image)
+
+                    # Increment request sequence number
+                    responseSeqNum += 1
```
Comment (marc19997) [inline src/edge/modules/grpcExtension/app/inference_server.py:171]: This should always be incremented

**Pull request closed without merging**